File: src/app/conjugation/forms/imperative-negative.ts
import {Conjugation, Word, WordType} from "../conjugation";

export class ImperativeNegative implements Conjugation {
    getConjugation(word: Word): (Word | undefined) {
        switch (word.wordType) {

            // Verbs
            case WordType.GodanVerb:
            case WordType.IchidanVerb:
            case WordType.SuruVerb:
            case WordType.KuruVerb:
                return word.addSuffix('な');


            default:
                return undefined;
        }
    }

    getTitle(): string {
        return "Imperative, negative"
    }

}

File: backend/app/conjugation/forms/imperative_negative.py
from ..core import Conjugation, Word, WordType


class ImperativeNegative(Conjugation):

    title = 'Imperative, negative'
    settings_title = 'Negative'

    def conjugate(self, word: Word) -> Word | None:
        match word.word_type:

            # Verbs
            case (
                WordType.GODAN_VERB
                | WordType.ICHIDAN_VERB
                | WordType.SURU_VERB
                | WordType.KURU_VERB
            ):
                return word.add_suffix('な')

            case _:
                return None
